src/AppInstallerCLICore/Workflows/UninstallFlow.cpp

```diff
@@ -0,0 +1,201 @@
+            else if (context.Args.Contains(Execution::Args::Type::Silent))
+            {
+                itr = packageMetadata.find(PackageVersionMetadata::SilentUninstallCommand);
+            }
+            else
+            {
+                itr = packageMetadata.find(PackageVersionMetadata::StandardUninstallCommand);
+                if (itr == packageMetadata.end())
+                {
+                    itr = packageMetadata.find(PackageVersionMetadata::SilentUninstallCommand);
+                }
+            }
+
+            if (itr == packageMetadata.end())
+            {
+                AICLI_TERMINATE_CONTEXT(E_ABORT);
+            }
+            context.Add<Execution::Data::UninstallString>(itr->second);
+
+            break;
+        }
+        case ManifestInstaller::InstallerTypeEnum::Msix:
+        case ManifestInstaller::InstallerTypeEnum::MSStore:
+        {
+            context.Add<Execution::Data::PackageFamilyNames>(installedPackageVersion->GetMultiProperty(PackageVersionMultiProperty::PackageFamilyName));
+            break;
+        }
+        default:
+            THROW_HR(HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED));
+        }
+    }
+
+    void ExecuteUninstaller(Execution::Context& context)
+    {
+        const std::string installedTypeString = context.Get<Execution::Data::InstalledPackageVersion>()->GetMetadata()[PackageVersionMetadata::InstalledType];
+        switch (ManifestInstaller::ConvertToInstallerTypeEnum(installedTypeString))
+        {
+        case ManifestInstaller::InstallerTypeEnum::Exe:
+        case ManifestInstaller::InstallerTypeEnum::Burn:
+        case ManifestInstaller::InstallerTypeEnum::Inno:
+        case ManifestInstaller::InstallerTypeEnum::Msi:
+        case ManifestInstaller::InstallerTypeEnum::Nullsoft:
+        case ManifestInstaller::InstallerTypeEnum::Wix:
+            context << Workflow::ExecuteUninstallString;
+            break;
+        case ManifestInstaller::InstallerTypeEnum::Msix:
+            context << Workflow::MsixUninstall;
+            break;
+        case ManifestInstaller::InstallerTypeEnum::MSStore:
+            context <<
+                Workflow::EnsureFeatureEnabled(Settings::ExperimentalFeature::Feature::ExperimentalMSStore) <<
+                Workflow::MsixUninstall;
+            break;
+        default:
+        THROW_HR(HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED));
+        }
+    }
+
+    void ExecuteUninstallString(Execution::Context& context)
+    {
+        context.Reporter.Info() << Resource::String::UninstallFlowStartingPackageUninstall << std::endl;
+
+        auto uninstallResult = context.Reporter.ExecuteWithProgress(
+            std::bind(InvokeCreateProcess,
+                context.Get<Execution::Data::UninstallString>(),
+                std::placeholders::_1));
+
+        if (!uninstallResult)
+        {
+            // TODO: log & report
+            AICLI_TERMINATE_CONTEXT(E_ABORT);
+        }
+        else if (uninstallResult.value() != 0)
```

Comment:
>(uninstallResult.value() != 0) [](start = 16, length = 30)

This may need to be smarter; at a minimum I would expect a lot of "reboot required" errors coming out.  I think that for at least MSI we should probably figure out what can come back from msiexec.exe and be able to handle those values.

Reply:
I'm leaving this as a TODO for now as I'm not entirely sure what's the right thing to do with other exit codes. I'm trying to find an application that returns something non-zero to see what it does and how it is handled when uninstalling with Porgrams and Features or the Settings app. I'll probably add a telemetry point for errors that will help here.